ccm_web/client/src/components/BulkEnrollUMUserToSectionsConfirmationTable.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { useState } from 'react'
+
+import CustomTable from './CustomTable'
+import { RowNumberedAddEnrollmentWithSectionId } from '../models/enrollment'
+
+interface BulkEnrollUMUserSectionsConfirmationTableProps {
+  enrollments: RowNumberedAddEnrollmentWithSectionId[]
+}
+
+interface TableHeaderColumnInfoShouldUseMatUIType {
+  id: keyof RowNumberedAddEnrollmentWithSectionId
+  label: string
+  minWidth: number
+  align?: 'left' | 'right' | undefined
+}
+
+const columns: TableHeaderColumnInfoShouldUseMatUIType[] = [
+  { id: 'rowNumber', label: 'Row Number', minWidth: 25 },
+  { id: 'loginId', label: 'Login ID', minWidth: 100 },
+  { id: 'role', label: 'Role', minWidth: 100 },
+  { id: 'sectionId', label: 'Section ID', minWidth: 100 }
+]
+
+function BulkEnrollUMUserSectionsConfirmationTable (props: BulkEnrollUMUserSectionsConfirmationTableProps): JSX.Element {
+  const [page, setPage] = useState<number>(0)
+  const tableRows = props.enrollments
```

Comment:
I think some of these patterns could be further refactored and generalized. With a little tinkering, we could move the page state into `CustomTable` and then use that component directly (i.e. prepping `columns` and `tableRows` outside) rather than creating all these separate `{specificName}Table` components. Or, if we like the wrapper layer, we could add a generic to these wrappers and move the columns interface into `CustomTable`.

Reply:
https://github.com/tl-its-umich-edu/canvas-course-manager-next/issues/314